wallets/provider-xdefi/src/index.ts

```diff
@@ -50,7 +53,33 @@ export const connect: Connect = async ({ instance, meta }) => {
     meta,
   });
 
-  return [evmResult, ...nonEvmResults, solanaAccounts as ProviderConnectResult];
+  const cosmosAccounts = [];
+  if (cosmosInstance) {
+    const cosmosBlockchainMeta = meta
+      .filter(isCosmosBlockchain)
+      .filter((blockchainMeta: BlockchainMeta) =>
+        SUPPORTED_COSMOS_CHAINS.includes(blockchainMeta.name as Networks)
+      );
```

Comment:
Could you combine these filters and eliminate unnecessary iteration?

Reply:
Thanks Fixed